### PR TITLE
Add downgrade test

### PR DESCRIPTION
**Pull request by Sbozzolo**
To check for lower bound compats

### CODE REVIEW

File: .github/workflows/Downgrade.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        version: ['1']
+    steps:
+      - uses: actions/checkout@v4
+      - uses: julia-actions/setup-julia@latest
+        with:
+          version: ${{ matrix.version }}
+      - uses: julia-actions/cache@v2
+      - uses: julia-actions/julia-downgrade-compat@v1
+        with:
+          skip: Artifacts, Dates, Profile, Test
```
Comment (juliasloan25) [inline .github/workflows/Downgrade.yml:32]: I don't see Profile in the top-level or test Projects, maybe we can remove it here?

**Review by juliasloan25 — APPROVED** [review]
Thanks Gabriele, this looks super helpful! I think it would be helpful to have this in ClimaLand and ClimaCoupler too, so I may add it there

File: .github/workflows/Downgrade.yml (hunk 1 above)
Comment (Sbozzolo) [reply .github/workflows/Downgrade.yml:32]: You are right. Thank you!

**Comment by Sbozzolo** [comment]
> Thanks Gabriele, this looks super helpful! I think it would be helpful to have this in ClimaLand and ClimaCoupler too, so I may add it there

FYI, it may be painful to set up because you have to find the correct lower bounds for the Project.toml (and the only way I know how to is with trial and error)

**Pull request merged**